docs/gt4py/quickstart.rst

```diff
@@ -183,29 +185,43 @@ set of routines used in `NumPy` for allocation: ``from_array``, ``ones``, ``zero
         data=np.random.randn(10, 10, 10),
         backend=backend,
         dtype=np.float64,
-        default_origin=(0, 0, 0),
+        aligned_index=(0, 0, 0),
     )
     field_b = gt_storage.ones(
-        backend=backend, shape=(10, 10, 10), dtype=np.float64, default_origin=(0, 0, 0)
+        backend=backend, shape=(10, 10, 10), dtype=np.float64, aligned_index=(0, 0, 0)
     )
     field_c = gt_storage.zeros(
-        backend=backend, shape=(10, 10, 10), dtype=np.float64, default_origin=(0, 0, 0)
+        backend=backend, shape=(10, 10, 10), dtype=np.float64, aligned_index=(0, 0, 0)
     )
     result = gt_storage.empty(
-        backend=backend, shape=(10, 10, 10), dtype=np.float64, default_origin=(0, 0, 0)
+        backend=backend, shape=(10, 10, 10), dtype=np.float64, aligned_index=(0, 0, 0)
     )
 
     stencil_example(field_a, field_b, field_c, result, alpha=0.5)
 
 
-The ``default_origin`` parameter plays two roles:
+The ``aligned_index`` specifies that the array is to be allocated such that memory address of the point specified in
+``aligned_index`` is `aligned` to a backend-dependent value. For optimal performance, you set the ``algined_index`` to
+a point which is the lower-left corner of the iteration domain most frequently used for this field.
```

Comment:
Still diving in here so apologies if this is clear later: does specifying `aligned_index` instead of `default_origin` mean that we always need to specify non-zero origins when calling stencils, or is this used as the default origin too?

Reply:
Kind of. If you just rely on the allocation routines, you get an unmodified NumPy ndarray. That is, there is no `default_origin` attribute like there used to be on the storages. However, if a user passes a buffer object that (This could e.g. be a `Quantity` in pace) They can opt to implenet an dunder attribute `__gt_dims__` that has then that functionality.